sklearn/preprocessing/tests/test_data.py

```diff
@@ -189,19 +209,338 @@ def test_scaler_2d_arrays():
     scaler = StandardScaler()
     X_scaled = scaler.fit(X).transform(X, copy=True)
     assert_false(np.any(np.isnan(X_scaled)))
-    assert_array_almost_equal(X_scaled.mean(axis=0), 5 * [0.0])
+    assert_array_almost_equal(X_scaled.mean(axis=0), n_features * [0.0])
     assert_array_almost_equal(X_scaled.std(axis=0), [0., 1., 1., 1., 1.])
     # Check that X has not been copied
     assert_true(X_scaled is not X)
 
 
+def test_incremental_mean_and_var():
+    # Test the helper for partial_fit()
+    for axis in [0, 1]:
+        rng = np.random.RandomState(0)
+        n_features = 50
+        offsets = rng.uniform(-1, 1, size=n_features)
+        scales = rng.uniform(1, 10, size=n_features)
+        chunk_size = 20
+        data_chunks_1 = rng.randn(chunk_size, n_features) * scales + offsets
+
+        # Test _incr_mean_and_var with a 1 row input
+        X = np.atleast_2d(np.array(data_chunks_1[0]))
+        # last_var=None
+        mean_incr, var_incr, _ = _incremental_mean_and_var(X, axis)
+
+        assert_almost_equal(np.mean(X, axis), mean_incr)
+        assert_equal(None, var_incr)
+
+        # last_var=.0
+        mean_incr, var_incr, _ = \
+            _incremental_mean_and_var(X, axis, last_var=.0)
+
+        assert_almost_equal(np.mean(X, axis), mean_incr)
+        if axis == 0:
+            assert_almost_equal(np.array([0 for i in range(n_features)]),
+                                var_incr)
+        else:
+            assert_almost_equal(np.var(X), var_incr)
+
+        # Test _incremantal_mean_and_var with whole data
+        X = np.vstack(data_chunks_1)
+        mean_incr, var_incr, _ = \
+            _incremental_mean_and_var(X, axis, last_var=.0)
+
+        assert_almost_equal(np.mean(X, axis), mean_incr)
+        assert_array_almost_equal(np.var(X, axis), var_incr)
+
+
+def test_incremental_min_and_max():
+    # Test the helper function for partial_fit
+    for axis in [0, 1]:
+        rng = np.random.RandomState(0)
+        n_features = 50
+        offsets = rng.uniform(-1, 1, size=n_features)
+        scales = rng.uniform(1, 10, size=n_features)
+        chunk_size = 10
+        data_chunks = rng.randn(chunk_size, n_features) * scales + offsets
+
+        # Test _incremental_min_and_max with a 1 row input
+        X = np.atleast_2d(np.array(data_chunks))
```

Comment:
Actually this weird I an not sure what this intention is here:

```python
>>> import numpy as np
>>> rng = np.random.RandomState(0)
>>> n_features = 50
>>> chunk_size = 10
>>> offsets = rng.uniform(-1, 1, size=n_features)
>>> scales = rng.uniform(1, 10, size=n_features)
>>> data_chunks = rng.randn(chunk_size, n_features) * scales + offsets
>>> data_chunks.shape
(10, 50)
>>> X = np.atleast_2d(np.array(data_chunks))
>>> X.shape
(10, 50)
```

Better generate `X` as a single block explicitly and use `gen_batches` to iterate over batches of various sizes.